node.gyp

```diff
@@ -177,7 +190,83 @@
   'targets': [
     {
       'target_name': '<(node_core_target_name)',
-      'type': '<(node_target_type)',
+      'type': 'executable',
+      'dependencies': [
+        '<(node_lib_target_name)'
+      ],
+     'sources': [
+        'src/node_main.cc'
+      ],
+      'include_dirs': [
+        'src',
+        'deps/v8/include',
+      ],
+      'conditions': [
+        [ 'node_intermediate_lib_type=="static_library" and '
+            'node_shared=="true" and OS=="aix"', {
+           # For AIX, shared lib is linked by static lib and .exp. In the
+           # case here, the executable needs to link to shared lib.
+           # Therefore, use 'node_aix_shared' target to generate the
+           # shared lib and then executable.
+           'dependencies!': [ '<(node_lib_target_name)' ],
+           'dependencies': [ 'node_aix_shared' ],
+        }],
```

Comment:
I think it's clearer if you drop the `'dependencies!'` and add an alternate, i.e.:
```python
'conditions': [
  ['...', {
    'dependencies': ['node_aix_shared'],
  }, {
    'dependencies': ['<(node_lib_target_name)'],
  }],
```

Reply:
fixed